File: app/src/main/java/mishka/rssreader/data/rss/RetrofitRssFetcher.java
package mishka.rssreader.data.rss;

import org.simpleframework.xml.convert.AnnotationStrategy;
import org.simpleframework.xml.core.Persister;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import mishka.rssreader.data.model.RealmRss;
import mishka.rssreader.data.model.RealmRssItem;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.simplexml.SimpleXmlConverterFactory;
import retrofit2.http.GET;

public class RetrofitRssFetcher implements RssHelper {

    @Inject
    public RetrofitRssFetcher() {
    }

    @Override
    public List<RealmRssItem> getRealmRss() {
        Call<RealmRss> realmCall = getVestiRssApi().getRealmRss();
        List<RealmRssItem> realmItems = new ArrayList<>();

        try {
            realmItems = realmCall.execute().body().getChannel().getItems();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return realmItems;
    }

    private VestiRssApi getVestiRssApi() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://www.vesti.ru/")
                .addConverterFactory(
                        SimpleXmlConverterFactory.createNonStrict(
                                new Persister(new AnnotationStrategy())
                        )
                )
                .build();

        return retrofit.create(VestiRssApi.class);
    }

    interface VestiRssApi {
        @GET("vesti.rss")
        Call<RealmRss> getRealmRss();
    }
}

File: app/src/main/java/mishka/rssreader/data/rss/RetrofitRssFetcher.kt
package mishka.rssreader.data.rss

import javax.inject.Inject

import mishka.rssreader.data.model.Rss
import mishka.rssreader.data.model.RssItem
import mishka.rssreader.ui.channelsettings.ChannelConfigUtils
import org.simpleframework.xml.convert.AnnotationStrategy
import org.simpleframework.xml.core.Persister
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.simplexml.SimpleXmlConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Url
import java.io.IOException

class RetrofitRssFetcher @Inject constructor(private val channelConfigUtils: ChannelConfigUtils) : RssHelper {

    override fun getRss(): List<RssItem>? {
        val retrofit = Retrofit.Builder()
                .baseUrl("https://stub/")
                .addConverterFactory(
                        SimpleXmlConverterFactory.createNonStrict(
                                Persister(AnnotationStrategy())
                        )
                )
                .build()

        val vestiRssApi: VestiRssApi = retrofit.create(VestiRssApi::class.java)
        val call = vestiRssApi.getRss(channelConfigUtils.getCurrentChannelUrl())

        var items: List<RssItem>? = null
        try {
            items = call.execute().body()?.channel?.items
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return items
    }

    interface VestiRssApi {
        @GET
        fun getRss(@Url url: String): Call<Rss>
    }


}
